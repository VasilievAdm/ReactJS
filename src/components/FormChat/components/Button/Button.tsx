import React, { FC } from 'react';

interface ButtonProps {
  click: () => void;
  name: string;
}

export const Button: FC<ButtonProps> = (props) => {
  return (
    <button
      className="button"
      onClick={props.click}
      style={{ backgroundColor: 'red' }}
    >
      {props.name}
    </button>
  );
};
