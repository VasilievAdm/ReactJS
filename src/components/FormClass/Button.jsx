import React, { Component } from 'react';

export class Button extends Component {
  render() {
    return <button className='button' onClick={this.props.click}>{this.props.name}</button>
  }
}
