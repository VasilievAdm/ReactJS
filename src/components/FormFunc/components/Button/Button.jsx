import React from 'react';

export const Button = (props) => {
  return (
    <button
      className="button"
      onClick={props.click}
      disabled={props.disabled}
      style={{ backgroundColor: 'red' }}
    >
      {props.name}
    </button>
  );
};
// import React from 'react';

// export const Button = ({ disabled, onButtonClick }) => {
//   return (
//     <button type="submit" disabled={disabled} onClick={onButtonClick}>
//       click
//     </button>
//   );
// };
