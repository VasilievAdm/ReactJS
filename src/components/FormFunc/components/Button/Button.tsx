import React, { FC } from 'react';
import { Button as ButtonUI } from '@mui/material';

interface ButtonProps {
  disabled: boolean;
  click: () => void;
  name: string;
}

export const Button: FC<ButtonProps> = (props) => {
  return (
    //   <button
    //     className="button"
    //     onClick={props.click}
    //     disabled={props.disabled}
    //     style={{ backgroundColor: 'red' }}
    //   >
    //     {props.name}
    //   </button>
    // );
    <ButtonUI onClick={props.click} disabled={props.disabled}>
      {props.name}
    </ButtonUI>
  );
};
