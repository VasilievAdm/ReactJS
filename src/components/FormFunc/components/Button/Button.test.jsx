import React from 'react';
import { Button } from './Button';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';
import { waitFor } from '@storybook/testing-library';
describe('Button', () => {
  it('render component', () => {
    render(<Button />);
  });

  it('render with snapshot', () => {
    const { asFragment } = render(<Button />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('render component with text', () => {
    render(<Button name="Send" />);
    expect(screen.getByText(/Send/)).toBeInTheDocument();
  });

  it('render multiply components', () => {
    render(
      <>
        <Button />
        <Button />
      </>
    );

    expect(screen.queryAllByRole('button').length).toBe(2);
  });

  it('button is disabled', () => {
    render(<Button disabled name="click" />);

    expect(screen.getByText('click')).toBeDisabled();
  });

  it('button have style background red', () => {
    render(<Button name="click" />);

    expect(screen.getByText('click')).toHaveStyle({ backgroundColor: 'red' });
  });

  it('button click with userEvent', async () => {
    const mockHandler = jest.fn();

    render(<Button click={mockHandler} name="click" />);
    await userEvent.click(screen.getByText(/click/));
    expect(mockHandler).toBeCalledTimes(1);
  });

  it('button async click', async () => {
    const mockHandler = jest.fn();

    render(<Button click={() => setTimeout(mockHandler, 1000)} name="click" />);

    await userEvent.click(screen.getByText(/click/));

    await waitFor(() => expect(mockHandler).toHaveBeenCalledTimes(1), {
      timeout: 1100,
    });
  });

  // it('test exapmle', async () => {
  //   const onChange = jest.fn();
  //   render(<input type="checkbox" onChange={onChange} />);
  //   const checkbox = screen.getByRole('checkbox');
  //   await userEvent.dblClick(checkbox);
  //   expect(onChange).toHaveBeenCalledTimes(2);
  //   expect(checkbox).not.toBeChecked();
  // });
});
