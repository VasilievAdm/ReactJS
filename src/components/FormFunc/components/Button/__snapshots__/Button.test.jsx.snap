// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button render with snapshot 1`] = `
<DocumentFragment>
  <button
    class="button"
    style="background-color: red;"
  />
</DocumentFragment>
`;
